pkg/app/server/grpcapi/api.go

```diff
@@ -533,6 +537,20 @@ func (a *API) GetPlanPreviewResults(ctx context.Context, req *apiservice.GetPlan
 		if cmd.Type != model.Command_BUILD_PLAN_PREVIEW {
 			return nil, status.Error(codes.FailedPrecondition, fmt.Sprintf("Command %s is not a plan preview command", commandID))
 		}
+		pipedStatus, err := getPipedStatus(a.pipedStatCache, cmd.PipedId)
+		if err != nil {
+			a.logger.Error("failed to get or unmarshal piped stat", zap.Error(err))
+			pipedStatus = model.Piped_UNKNOWN
+		}
+
+		if pipedStatus != model.Piped_ONLINE {
+			results = append(results, &model.PlanPreviewCommandResult{
+				CommandId: cmd.Id,
+				PipedId:   cmd.PipedId,
+				Error:     "Maybe Piped is offline currently.",
+			})
+			continue
+		}
```

Comment:
This check should be done inside the `if !cmd.IsHandled() {` block below.
Because a command could be already handled successfully before the Piped goes offline. In that case, the command result is still valid.

Reply:
That's right. Thank you!